### PR TITLE
[#11639] Create DateTime service and replace time-related magic numbers

### LINKED ISSUE #11639: Refactor date and time methods into utility file
The frontend uses DateFormat/TimeFormat, Moment and epoch timestamps, and even Date.
However, functions to parse and manipulate between the data types are scattered in different files and folders.

Examples that can benefit from refactoring:
- Put methods to convert between the types into one single utility file
E.g getDateTimeAtTimezone (conversion from timestamp to date/time format) implemented only in Session Edit page
- Time-related magic numbers being declared locally repeatedly, or even just used plainly as numbers throughout the repo

### PR DESCRIPTION
**Pull request by tau-bar**
Fixes #11639 

**Outline of Solution**
1. I created a constants file for common time-related constants, to remove magic numbers throughout the project. I then used these constants to replace the magic numbers around the project.
2. Moved `DateFormat` and `TimeFormat` into constants file.
3. I created a `DateTimeService` to store helper functions relating to datetime.

### CODE REVIEW

File: src/web/app/pages-monitoring/logs-page/logs-page.component.ts
Comment (FergusMok) [inline src/web/app/pages-monitoring/logs-page/logs-page.component.ts:60]: Could convert this to milliseconds in one day instead?

File: src/web/types/datetime-const.ts
Comment (FergusMok) [inline src/web/types/datetime-const.ts:18]: I'm thinking the current format `MsConst.ONE_SECOND_MILLISECONDS` may not be immediately obvious when used.

Would something like `Milliseconds.IN_ONE_SECOND` be clearer?

**Review by FergusMok — CHANGES_REQUESTED** [review]
Thanks @tau-bar for the refactoring, just need you to make some small changes

**Comment by tau-bar** [comment]
@FergusMok I've updated based on the requested changes, could you help me review the code again? Thanks!

File: src/web/services/datetime.service.ts
Comment (FergusMok) [inline src/web/services/datetime.service.ts:47]: I tried looking for the `@code` tag here: https://tsdoc.org/ and I couldnt find anything :/ 
My IDE is not processing it either for typescript, and I think it might be a purely Java doc convention that other authors brought to the Typescript frontend

**Comment by FergusMok** [comment]
@tau-bar Thanks for the work so far! Definitely see the codebase being cleaner now

File: src/web/services/datetime.service.spec.ts
Comment (FergusMok) [reply src/web/services/datetime.service.spec.ts:46]: And also one with both hours and minutes out of bounds

File: src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-date-modal/individual-extension-date-modal.component.ts
Comment (FergusMok) [inline src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-date-modal/individual-extension-date-modal.component.ts:71]: Could you help replace these? Shouldn't need these anymore with the new constants.

**Comment by FergusMok** [comment]
@tau-bar I'm ok with the PR after the last change and fixing the E2E tests! Will wait for one more review

**Comment by tau-bar** [comment]
@FergusMok Thanks for all the help so far! I've removed the constants as requested.

**Review by FergusMok — APPROVED** [review]

File: src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts
Comment (zhaojj2209) [inline src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts:219]: This line seems to be left out of the refactoring, is it intentional?

**Review by zhaojj2209 — COMMENTED** [review]
Good work with the refactoring! One last nit

**Comment by FergusMok** [comment]
@tau-bar Just a small reminder about the PR!

**Comment by tau-bar** [comment]
> @tau-bar Just a small reminder about the PR!

Sorry, got caught up in other things. Will continue working on both of my PRs next week!

**Comment by fsgmhoward** [comment]
@samuelfangjw @ziqing26 @zhaojj2209 Any review for this PR would be appreciated.

File: src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts
Comment (zhaojj2209) [reply src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.spec.ts:219]:
```suggestion
    submissionStartDate: getDefaultDateFormat(),
```
@tau-bar Do make the above suggested change!

**Review by ziqing26 — APPROVED** [review]
LGTM! 👍 Other than please fix the nitpick that @zhaojj2209 mentioned:)

**Comment by tau-bar** [comment]
Hello @ziqing26 @zhaojj2209 I've changed based on the comments given, sorry it took so long!

**Review by zhaojj2209 — APPROVED** [review]
LGTM, thanks for the changes!

**Pull request merged**
